ironfist/src/cpp/game/save.cpp

```diff
@@ -228,28 +228,38 @@ ironfist_map::hero_t WriteHeroXML(hero* hro) {
 	return hx;
 }
 
+void SetMapVariableValue(ironfist_map::mapVariable_t &mapVar, luaTable lt) {
+	for (luaTable::const_iterator it = lt.begin(); it != lt.end(); ++it) {

```

Comment:
I could tell this code was better than previous in an instant.

Notice the simple control flow, and how the control flow mirrors the structure of the data. When you update the map_xml so that tables can be recursive, this will correspondingly become a Lua function.